### PR TITLE
[skrifa] autohint: prep for CJK

### PR DESCRIPTION
**Pull request by dfrg**
Just adds a CJK test font, a test for CJK widths (exact same computation as Latin), and shuffles the current Latin blues code around to make way for the CJK stuff.

No functional changes.

Refactoring the Latin blues code from method to free function generated a ton of noise in the diff so this was pulled out to make the CJK blues work actually reviewable without the noise.

### CODE REVIEW

File: font-test-data/test_data/ttx/notoseriftc_autohint_metrics.ttx
Comment (rsheeter) [inline font-test-data/test_data/ttx/notoseriftc_autohint_metrics.ttx:1]: I'm belatedly realizing we've done a really bad job documenting where our ttx files came from. Suggest adding a bullet to ... maybe font-test-data/test_data/README.md ... saying how to produce this file, e.g. get the whole binary from repo, subset using _command_ and ttx

File: skrifa/src/outline/autohint/latin/widths.rs [hunk 1]
```diff
@@ -134,6 +134,32 @@ mod tests {
         );
     }
 
+    #[test]
+    fn cjk_computed_widths() {
+        check_widths(
+            font_test_data::NOTOSERIFTC_AUTOHINT_METRICS,
+            super::ScriptClass::HANI,
+            [
```
Comment (rsheeter) [inline skrifa/src/outline/autohint/latin/widths.rs:142]: suggest comment explaining where the expected value scome from. FT I assume, but how do you actually create the data? Or is it just what we observed our code to produce?

**Review by rsheeter — APPROVED** [review]

File: font-test-data/test_data/ttx/notoseriftc_autohint_metrics.ttx
Comment (rsheeter) [reply font-test-data/test_data/ttx/notoseriftc_autohint_metrics.ttx:1]: Context https://github.com/googlefonts/fontations/issues/1127, it'll make our lives easier to start capturing the info ^

Comment (dfrg) [reply font-test-data/test_data/ttx/notoseriftc_autohint_metrics.ttx:1]: Added a section for this to the test_data readme with details about the font source, license, usage and command used to subset

File: skrifa/src/outline/autohint/latin/widths.rs (hunk 1 above)
Comment (dfrg) [reply skrifa/src/outline/autohint/latin/widths.rs:148]: Noted that these values are scraped from a FT debugging session.

In this case, I just copied the test skeleton and manually transcribed the values from the debugger. I figured the odds of me making a transcription error that matches a bug are vanishingly low :) For tests with more data (segments, edges, etc), I have some ugly printf code in my local FT copy that prints the expected values to the console and I copy that into the tests. This is really handy in cases where FT uses pointers but we use indices.

I tend to only do "regression style" tests (copying in values that we produce) when we already have known good tests to start with.

**Pull request merged**